src/scripts/main.js

```diff
@@ -1,11 +1,19 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
+      );
+    });
+  });
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const body = document.querySelector('body');
+
+  body.insertAdjacentHTML('beforeend', `<div class="message">${message}</div>`);
```

Comment:
Directly inserting HTML into the body can be a bad practice, particularly for larger applications. It would be better to have a dedicated container for messages. This would allow you to control the placement and styling of messages more easily.